File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
       unordered_map<char,int> mpp;
        int l =0;
        int r =0;
        int maxi =0, ct =0;
        while( r  < s.length() ) {
            mpp[s[r]]++;

            while( mpp[s[r]] > 1 ) {
                mpp[s[l]]--;
                // if( mpp[s[l]] == 1 ) {
                //     mpp.erase(s[r]);
                // }
                l++;
            }
           maxi = max(maxi, r - l + 1);
            r++;
        }

      return maxi;
    }
};

File: Data Structures & Algorithms/find-lucky-integer-in-an-array/submission-1.java
class Solution {
    public int findLucky(int[] arr) {
        

        HashMap<Integer,Integer> mpp = new HashMap<>();

        int maxi = -1;
        for( int i=0; i<arr.length; i++ ) {
            mpp.put( arr[i] , mpp.getOrDefault(arr[i] , 0 ) + 1);
        }

        for( int i=0; i<arr.length; i++ ) {
            if( mpp.get(arr[i]) == arr[i] ) {
                 maxi = Math.max( maxi , arr[i]);
            }
        }

        return maxi;
    }
}